addons/mrp/models/mrp_workorder.py

```diff
@@ -489,9 +489,9 @@ def write(self, values):
         if 'workcenter_id' in values:
             new_workcenter = self.env['mrp.workcenter'].browse(values['workcenter_id'])
             for workorder in self:
-                if workorder.workcenter_id.id != values['workcenter_id']:
-                    if workorder.state in ('progress', 'done', 'cancel'):
-                        raise UserError(_('You cannot change the workcenter of a work order that is in progress or done.'))
+                if workorder.workcenter_id.id != values['workcenter_id'] and workorder.state != 'progress':
```

Comment:
```suggestion
                if workorder.workcenter_id.id != values['workcenter_id']
```
Put the `workorder.state != 'progress'` condition before the recomputation of the duration/dates if we want to ignore the recomputation that should be done.

But the workcenter's `resource_id` should still be set.